### PR TITLE
Support v0.3 schema report format in LLO (common)

### PR DESCRIPTION
**Pull request by samsondav**

### CODE REVIEW

File: pkg/types/llo/types_test.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+		}
+		b := ChannelDefinition{
+			ReportFormat: ReportFormatEVMPremiumLegacy,
+			StreamIDs:    []StreamID{0, 1},
+			Aggregators:  []Aggregator{AggregatorMedian, AggregatorMode},
+			Opts:         nil,
+		}
+		assert.False(t, a.Equals(b))
+	})
+	t.Run("different StreamIDs", func(t *testing.T) {
```
Comment (msuchacz-cll) [inline pkg/types/llo/types_test.go:73]: is this array always sorted? I noticed your equals function compares them by checking a corresponding index

**Review by msuchacz-cll — APPROVED** [review]
if StreamIDs array is always sorted then the rest LGTM

File: pkg/types/llo/types_test.go (hunk 1 above)
Comment (samsondav) [reply pkg/types/llo/types_test.go:73]: Yes ordering is considered to be a critical part of the channel definition. Different order is a different channel

File: pkg/types/llo/types.go [hunk 2]
```diff
@@ -95,14 +191,82 @@ type Transmitter ocr3types.ContractTransmitter[ReportInfo]
...
-	// We assume that StreamIDs is always non-empty and that the 0-th stream
-	// contains the verification price in LINK and the 1-st stream contains the
-	// verification price in the native coin.
-	StreamIDs []StreamID
+	ReportFormat ReportFormat `json:"reportFormat"`
+	// StreamIDs contains a list of streams to be observed for this channel.
+	StreamIDs []StreamID `json:"streamIDs"`
+	// Aggregators contains a list of aggregator methods used for each stream
+	// e.g. "median", "mode" or other more exotic methods.
+	// StreamIDs and Aggregators must have the same length.
```
Comment (akuzni2) [inline pkg/types/llo/types.go:199]: is there any limitation to why StreamIDs and Aggregators are split up? If they are a 1:1 pair wouldn't it just be easier + less error prone to define them together?
```
Things []Thing json:"things"`

type Thing struct {
   s StringId
   a Aggregator 
}

```

Comment (samsondav) [reply pkg/types/llo/types.go:199]: Original reason is that it maps more closely to how you'd represent that data on-chain, and it keeps all data types that may need to be interpreted on-chain simple.

Comment (samsondav) [reply pkg/types/llo/types.go:199]: It's a fair point though, it might be safer to use a tuple/composite type here.

**Review by akuzni2 — APPROVED** [review]

**Review by austinborn — APPROVED** [review]

**Review by akuzni2 — APPROVED** [review]

**Review by austinborn — DISMISSED** [review]

**Review by msuchacz-cll — APPROVED** [review]

**Review by Atrax1 — APPROVED** [review]

**Pull request merged**